### PR TITLE
Add workflow selection and tiny UI improvements

### PR DESCRIPTION
**Pull request by apennell**
- Adds an optional `workflowOptions` param to the config passed to
TranslationsTab from the studio, which accepts an array of workflows.
Workflows are expected to include a `workflowUid`, which can be found
in the Smartling project's UI, and a `workflowName` for display title.
- The workflow select input only shows if at least one workflow option
is passed, and otherwise the additional work of setting the workflow
and locales in task creation isn't done
- The selected workflowUid string (if present) is passed as a param
with the rest of the context to the adapter's `createTask` function
- This replaces the `isWorkflowMT` boolean, allowing user to pass in
one or more specific workflow uids and names.

Example of the config in deskStructure, using `workflowOptions`:
```
const myCustomConfig = {
  ...defaultFieldLevelConfig,
  workflowOptions: [
    { workflowUid: 'abc123', workflowName: 'Machine Translation (testing)' }
  ]
}
```

Also updates styling and layout to add more spacing between sections 
and use appropriate headers to more clearly delineate the form and the 
progress sections, and the sections within the form.

Screen Shot 2022-01-19 at 3 41 33 PM
Note: this screenshot is of a studio with custom theming, so if any colors look
off, that's likely why. I did make changes to some of the tones used where it
seemed appropriate, such as the locale labels in the toggle boxes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.1.44",
+  "version": "2.0.0",
```
Comment (cngonzalez) [inline package.json:2]:
```suggestion
  "version": "2.0.0",
  "version": "1.2.0"
```
I know I initially said we might have to go up a version, but I don't think this is a breaking change (and I have plans for 2.0.0 actually being a breaking change, so I don't want to cause too much noise). Think we can stay in the 1s!

File: src/components/NewTask.tsx [hunk 2]
```diff
@@ -1,7 +1,19 @@
-import React, { useState, useContext } from 'react'
+import React, { useState, useContext, ChangeEvent } from 'react'
+import styled from 'styled-components'
+import {
+  Button,
+  Box,
+  Flex,
+  Grid,
+  Heading,
```
Comment (cngonzalez) [inline src/components/NewTask.tsx:8]:
```suggestion
```
We tend not to use Headings in the studio context. I've replaced them with Text as h2s (understandable if this doesn't make sense with your styling, just trying to stay consistent with our own packages!)

File: src/components/NewTask.tsx [hunk 3]
```diff
@@ -33,11 +48,9 @@ const LocaleCheckbox = ({ locale, toggle, checked }: LocaleCheckboxProps) => {
           //not needed because of above toggle logic, but silence React warnings.
           onChange={() => {}}
         />
-        <Box>
-          <Text size={1} weight="semibold">
-            {locale.description}
-          </Text>
-        </Box>
+        <WrapText>
+          <Text size={1}>{locale.description}</Text>
```
Comment (cngonzalez) [inline src/components/NewTask.tsx:52]:
```suggestion
 <Text size={1} weight="semibold">{locale.description}</Text>
 ```
 
 A bit light in default styling, would like to keep it readable.

File: src/components/NewTask.tsx [hunk 4]
```diff
@@ -85,45 +100,77 @@ export const NewTask = ({ locales }: Props) => {
   const possibleLocales = locales.filter(locale => locale.enabled !== false)
 
   return (
-    <Stack space={3}>
-      <Flex align="center" justify="space-between">
-        <Text weight="semibold" size={1}>
-          {possibleLocales.length === 1 ? `Select locale` : `Select locales`}
-        </Text>
+    <Stack paddingTop={4} space={4}>
+      <Heading as="h2" weight="semibold" size={2}>
```
Comment (cngonzalez) [inline src/components/NewTask.tsx:104]:
```suggestion
      <Text as="h2" weight="semibold" size={2}>
```
See above comments on standards in `Heading` vs `Text`

File: src/components/NewTask.tsx [hunk 5]
```diff
@@ -85,45 +100,77 @@ export const NewTask = ({ locales }: Props) => {
...
+        Create New Translation Job
+      </Heading>
```
Comment (cngonzalez) [inline src/components/NewTask.tsx:106]:
```suggestion
      </Text>
```

File: src/components/TaskView.tsx [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React, { useContext } from 'react'
-import { Box, Stack, Text, useToast } from '@sanity/ui'
+import { Box, Heading, Stack, useToast } from '@sanity/ui'
```
Comment (cngonzalez) [inline src/components/TaskView.tsx:2]:
```suggestion
import { Box, Text, Stack, useToast } from '@sanity/ui'
```
See above comments on standards in `Heading` vs `Text`

File: src/components/TaskView.tsx [hunk 7]
```diff
@@ -33,10 +33,10 @@ export const TaskView = ({ task, locales }: JobProps) => {
   }
 
   return (
-    <Stack space={3}>
-      <Text weight="semibold" size={1}>
-        Current job progress
-      </Text>
+    <Stack space={4}>
+      <Heading as="h2" weight="semibold" size={2}>
```
Comment (cngonzalez) [inline src/components/TaskView.tsx:37]:
```suggestion
      <Text as="h2" weight="semibold" size={2}>
```
See above comments on standards in Heading vs Text

File: src/components/TaskView.tsx [hunk 8]
```diff
@@ -33,10 +33,10 @@ export const TaskView = ({ task, locales }: JobProps) => {
...
+        Current Job Progress
+      </Heading>
```
Comment (cngonzalez) [inline src/components/TaskView.tsx:39]:
```suggestion
      </Text>
```

**Review by cngonzalez — CHANGES_REQUESTED** [review]
This looks great! I made annoying fiddly suggestions just based on default studio styling -- let me know if that works for you.

**Comment by apennell** [comment]
Those suggestions are fine and I went ahead and applied them. Thanks!

File: src/components/NewTask.tsx (hunk 4 above)
Comment (cngonzalez) [reply src/components/NewTask.tsx:103]: Sorry, I don't think the <Text> suggestion made it into the code -- would you mind applying it?

File: src/components/NewTask.tsx (hunk 3 above)
Comment (cngonzalez) [reply src/components/NewTask.tsx:51]: Sorry I don't think the "semibold" suggestion made it into the code, would you mind applying it?

File: package.json (hunk 1 above)
Comment (cngonzalez) [reply package.json:2]: I messed up my suggesting by including 2 versions :/ would you mind applying this new one?

**Review by cngonzalez — CHANGES_REQUESTED** [review]
I think some of the suggestions didn't make it through! I put comments where that happened.

**Comment by apennell** [comment]
Sorry about that @cngonzalez! First time accepting suggestions in GH and probably skipped a step. Should be good now, though.

**Review by cngonzalez — APPROVED** [review]

**Pull request merged**